src/CookieActionComponent.cpp

```diff
@@ -56,21 +59,27 @@ CookieThrower::CookieThrower() {
             glm::vec3(0.1),
             NULL,
             NULL,
-            bunnyRenderComp2);
+            bunnyRenderComp2,
+            NULL);
+
+
+}
+
+CookieActionComponent::~CookieActionComponent() {}
+
+void CookieActionComponent::initActionComponent() {
 
     GameManager::instance().getGameWorld().addDynamicGameObject(gameObj);
     GameManager::instance().getGameWorld().addDynamicGameObject(gameObj1);
     GameManager::instance().getGameWorld().addDynamicGameObject(gameObj2);
 
 }
 
-CookieThrower::~CookieThrower() {}
-
-void CookieThrower::pollAndThrow(double deltaTime, double totalTime) {
+void CookieActionComponent::checkAndPerformAction(double deltaTime, double totalTime) {
 
     GameManager& gameManager = GameManager::instance();
     Camera& camera = gameManager.getCamera();
-    GameObject& player = gameManager.getPlayer();
+    //GameObject& player = gameManager.getPlayer();
```

Comment:
I'm about to remove this in my next PR so this is fine. Just want to comment for the sake of documentation.